### PR TITLE
[13.0] [MIG] quality_control_oca_stock: Migration to v13.0

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT26775

This PR superseds https://github.com/OCA/manufacture/pull/483 ping @ps-tubtim 

please review this @pedrobaeza @victoralmau

### CODE REVIEW

File: quality_control_oca_stock/i18n/ca.po [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Stock Move"
+#~ msgstr "Moure l'estoc"
+
+#~ msgid "Product"
+#~ msgstr "Producte"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/ca.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/de.po [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+#~ msgstr "Kommissionierschein"
+
+#~ msgid "Stock Move"
+#~ msgstr "Lagerbuchung"
+
+#~ msgid "The operation type determines the picking view"
+#~ msgstr "Der Vorgangstyp legt die Pick-Ansicht fest"
+
+#~ msgid "Product"
+#~ msgstr "Produkt"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/de.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/el_GR.po [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+msgstr ""
+
+#. module: quality_control_stock
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Product"
+#~ msgstr "Προϊόν"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/el_GR.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/es.po [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+#~ msgstr "Albarán"
+
+#~ msgid "Stock Move"
+#~ msgstr "Movimiento de stock"
+
+#~ msgid "The operation type determines the picking view"
+#~ msgstr "El tipo de operación determina la vista del albarán"
+
+#~ msgid "Product"
+#~ msgstr "Producto"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/es.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/es_AR.po [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Picking List"
+#~ msgstr "Remitos"
+
+#~ msgid "Stock Move"
+#~ msgstr "Movimiento de stock"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/es_AR.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/es_ES.po [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+msgstr ""
+
+#. module: quality_control_stock
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Product"
+#~ msgstr "Producto"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/es_ES.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/es_MX.po [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+msgstr ""
+
+#. module: quality_control_stock
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Stock Move"
+#~ msgstr "Movimiento bursátil"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/es_MX.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/eu.po [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+msgstr ""
+
+#. module: quality_control_stock
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Product"
+#~ msgstr "Produktua"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/eu.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/fi.po [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+msgstr ""
+
+#~ msgid "Picking List"
+#~ msgstr "Keräilylista"
+
+#~ msgid "Stock Move"
+#~ msgstr "Varastosiirto"
+
+#~ msgid "Product"
+#~ msgstr "Tuote"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/fi.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/fr.po [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+
+#~ msgid "Stock Move"
+#~ msgstr "Mouvement de stock"
+
+#, fuzzy
+#~ msgid "The operation type determines the picking view"
+#~ msgstr "Le type de préparation détermine la vue de préparation"
+
+#~ msgid "Product"
+#~ msgstr "Article"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/fr.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/fr_CH.po [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+msgstr ""
+
+#. module: quality_control_stock
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Product"
+#~ msgstr "Produit"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/fr_CH.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/fr_FR.po [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+msgstr ""
+
+#. module: quality_control_stock
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Product"
+#~ msgstr "Produit"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/fr_FR.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/gl.po [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Stock Move"
+#~ msgstr "Movemento de existencias"
+
+#~ msgid "Product"
+#~ msgstr "Produto"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/gl.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/hr.po [hunk 14]
```diff
@@ -0,0 +1,159 @@
...
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Stock Move"
+#~ msgstr "Skladišni prenos"
+
+#~ msgid "Product"
+#~ msgstr "Proizvod"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/hr.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/hr_HR.po [hunk 15]
```diff
@@ -0,0 +1,156 @@
...
+msgstr ""
+
+#. module: quality_control_stock
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Product"
+#~ msgstr "Proizvod"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/hr_HR.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/it.po [hunk 16]
```diff
@@ -0,0 +1,167 @@
...
+msgstr "ispezioni"
+
+#~ msgid "Picking List"
+#~ msgstr "Lista Picking"
+
+#~ msgid "Stock Move"
+#~ msgstr "Movimento di magazzino"
+
+#~ msgid "Product"
+#~ msgstr "Prodotto "
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/it.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/nl.po [hunk 17]
```diff
@@ -0,0 +1,155 @@
...
+msgstr ""
+
+#. module: quality_control_stock
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Product"
+#~ msgstr "Product"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/nl.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/nl_NL.po [hunk 18]
```diff
@@ -0,0 +1,155 @@
...
+msgstr ""
+
+#. module: quality_control_stock
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Product"
+#~ msgstr "Product"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/nl_NL.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/pt.po [hunk 19]
```diff
@@ -0,0 +1,158 @@
...
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Stock Move"
+#~ msgstr "Movimento Stock"
+
+#~ msgid "Product"
+#~ msgstr "Produto"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/pt.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/pt_BR.po [hunk 20]
```diff
@@ -0,0 +1,171 @@
...
+msgstr "Inspeções"
+
+#~ msgid "Picking List"
+#~ msgstr "Lista de separação"
+
+#~ msgid "Stock Move"
+#~ msgstr "Movimentação de estoque"
+
+#~ msgid "Product"
+#~ msgstr "Produto"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/pt_BR.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/ro.po [hunk 21]
```diff
@@ -0,0 +1,159 @@
...
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Stock Move"
+#~ msgstr "Mișcare stoc"
+
+#~ msgid "Product"
+#~ msgstr "Produs"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/ro.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/ru.po [hunk 22]
```diff
@@ -0,0 +1,157 @@
...
+msgstr ""
+
+#. module: quality_control_stock
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Product"
+#~ msgstr "Товар/Услуга"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/ru.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/sk.po [hunk 23]
```diff
@@ -0,0 +1,155 @@
...
+msgstr ""
+
+#. module: quality_control_stock
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Product"
+#~ msgstr "Produkt"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/sk.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/sl.po [hunk 24]
```diff
@@ -0,0 +1,168 @@
...
+
+#~ msgid "Stock Move"
+#~ msgstr "Premik zaloge"
+
+#, fuzzy
+#~ msgid "The operation type determines the picking view"
+#~ msgstr "Tip zbirnika določi prikaz zbirnika"
+
+#~ msgid "Product"
+#~ msgstr "Proizvod"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/sl.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/tr.po [hunk 25]
```diff
@@ -0,0 +1,162 @@
...
+msgstr ""
+
+#~ msgid "Picking List"
+#~ msgstr "Toplama Listesi"
+
+#~ msgid "Stock Move"
+#~ msgstr "Stok Taşı"
+
+#~ msgid "Product"
+#~ msgstr "Ürün"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/tr.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/tr_TR.po [hunk 26]
```diff
@@ -0,0 +1,162 @@
...
+msgstr ""
+
+#~ msgid "Picking List"
+#~ msgstr "Seçim listesi"
+
+#~ msgid "Stock Move"
+#~ msgstr "Stok hareketi"
+
+#~ msgid "Product"
+#~ msgstr "Ürün"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/tr_TR.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/vi_VN.po [hunk 27]
```diff
@@ -0,0 +1,163 @@
...
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Stock Move"
+#~ msgstr "Dịch chuyển kho"
+
+#~ msgid "Product"
+#~ msgstr "Sản phẩm"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/vi_VN.po]:
```suggestion
```

File: quality_control_oca_stock/i18n/zh_CN.po [hunk 28]
```diff
@@ -0,0 +1,155 @@
...
+msgstr ""
+
+#. module: quality_control_stock
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_lot_qc_view
+#: model_terms:ir.ui.view,arch_db:quality_control_stock.stock_picking_qc_view
+msgid "inspections"
+msgstr ""
+
+#~ msgid "Product"
+#~ msgstr "产品"
```
Comment (victoralmau) [inline quality_control_oca_stock/i18n/zh_CN.po]:
```suggestion
```

File: quality_control_oca_stock/models/qc_inspection.py [hunk 29]
```diff
@@ -0,0 +1,102 @@
...
+                        )
+                    ).lot_id
+                elif inspection.object_id._name == "stock.production.lot":
+                    inspection.lot_id = inspection.object_id
+
+    @api.depends("object_id")
+    def _compute_product_id(self):
+        """Overriden for getting the product from a stock move."""
+        self.ensure_one()
+        super(QcInspection, self)._compute_product_id()
```
Comment (victoralmau) [inline quality_control_oca_stock/models/qc_inspection.py]:
```suggestion
        super()._compute_product_id()
```

File: quality_control_oca_stock/models/stock_picking.py [hunk 30]
```diff
@@ -0,0 +1,76 @@
...
+            count_data = picking_data.get(picking.id, {})
+            picking.created_inspections = sum(count_data.values())
+            picking.passed_inspections = count_data.get("success", 0)
+            picking.failed_inspections = count_data.get("failed", 0)
+            picking.done_inspections = (
+                picking.passed_inspections + picking.failed_inspections
+            )
+
+    def action_done(self):
+        res = super(StockPicking, self).action_done()
```
Comment (victoralmau) [inline quality_control_oca_stock/models/stock_picking.py]:
```suggestion
        res = super().action_done()
```

File: quality_control_oca_stock/models/stock_picking_type.py [hunk 31]
```diff
@@ -0,0 +1,36 @@
...
+                "name": picking_type.name,
+                "company_id": picking_type.warehouse_id.company_id.id,
+                "picking_type_id": picking_type.id,
+                "partner_selectable": True,
+            }
+            self.env["qc.trigger"].sudo().create(qc_trigger)
+
+    @api.model_create_multi
+    def create(self, val_list):
+        picking_types = super(StockPickingType, self).create(val_list)
```
Comment (victoralmau) [inline quality_control_oca_stock/models/stock_picking_type.py]:
```suggestion
        picking_types = super().create(val_list)
```

File: quality_control_oca_stock/models/stock_picking_type.py [hunk 32]
```diff
@@ -0,0 +1,36 @@
...
+        picking_types._create_qc_trigger()
+        return picking_types
+
+    def write(self, vals):
+        res = super(StockPickingType, self).write(vals)
```
Comment (victoralmau) [inline quality_control_oca_stock/models/stock_picking_type.py]:
```suggestion
        res = super().write(vals)
```

File: quality_control_oca_stock/tests/test_quality_control_stock.py [hunk 33]
```diff
@@ -0,0 +1,357 @@
...
+# Copyright 2018 Simone Rubino - Agile Business Group
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import TransactionCase
+
+
+class TestQualityControl(TransactionCase):
+    def setUp(self):
+        super(TestQualityControl, self).setUp()
+
```
Comment (victoralmau) [inline quality_control_oca_stock/tests/test_quality_control_stock.py]:
```suggestion
```

File: quality_control_oca_stock/tests/test_quality_control_stock.py [hunk 34]
```diff
@@ -0,0 +1,357 @@
...
+            {
+                "name": "Lot for tests",
+                "product_id": self.product.id,
+                "company_id": self.company.id,
+            }
+        )
+        self.user1_id = self._create_user(
+            "user_1", [self.group_stock_user], self.company
+        )
+
```
Comment (victoralmau) [inline quality_control_oca_stock/tests/test_quality_control_stock.py]:
```suggestion
```

**Review by victoralmau — COMMENTED** [review]
Good job.

**Comment by CarlosRoca13** [comment]
@victoralmau changes done

**Review by victoralmau — APPROVED** [review]

**Comment by pedrobaeza** [comment]
I'm afraid the name is incorrect: it should be `quality_control_stock_oca`, not `quality_control_oca_stock`. The idea behind this is to not collapse with a possible `quality_control_stock` from Odoo by adding the suffix `_oca` at the end, and for not perverting the idea on `_oca` at the end for all the OCA modules that conflict with Odoo.

You can rename anyway with the same procedure, but you will need to change the PO again.

**Comment by CarlosRoca13** [comment]
@pedrobaeza changes done

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
And FYI, here is the counterpart in OpenUpgrade: https://github.com/OCA/OpenUpgrade/commit/fa311d8e0277fcf03d6c22f4e0200897fabe5a01

**Pull request merged**